### PR TITLE
Adds ClientSpanContext context manager

### PR DESCRIPTION
**Pull request by mjbryant**
This PR encompasses two issues: #22 and #23, as well as adds a new ClientSpanContext manager class, which allows the collection and logging of arbitrary trees of client spans. The motivation behind this new class is that we want to be able to group individual MySQL queries (`SELECT`, `BEGIN`) inside logical grouping that represent MySQL transactions. This solution is more general than any MySQL-specific attempt.

### CODE REVIEW

File: tests/logging_helper_test.py [hunk 1]
```diff
@@ -4,160 +4,131 @@
...
+            autospec=True)
+def test_zipkin_logging_context_log_spans(
+    copy_endpoint_mock, bin_ann_list_builder, ann_list_builder,
+    log_span_mock, time_mock
+):
+    # This lengthy function tests that the logging context properly
+    # logs both client and server spans, while attaching extra annotations
+    # logged throughout the context of the trace.
+    trace_id = '0xf'
+    parent_span_id = '0x1'
```
Comment (kaisen) [inline tests/logging_helper_test.py]: it's annoying, but can we use 16-char hex strings as test data here? it makes it more clear what kind of ids are expected

File: tests/logging_helper_test.py [hunk 2]
```diff
@@ -202,6 +183,25 @@ def test_zipkin_handler_raises_exception_if_ann_and_bann_not_provided(
             " for foo span" == str(excinfo.value))
 
 
+@mock.patch('pyramid_zipkin.logging_helper.thrift_obj_in_bytes', autospec=True)
+def test_log_span(thrift_obj):
+    # Not much logic here, so this is basically a smoke test
+    thrift_obj.return_value = 'obj'
+    registry = {'zipkin.transport_handler': (lambda x, y: (x, y))}
+    stream, span_bytes = logging_helper.log_span(
+        span_id='0x2',
```
Comment (kaisen) [inline tests/logging_helper_test.py]: same

**Comment by kaisen** [comment]
looks good to me. i assume you've tested this in dev?

let's tag my PR and this one together as v0.9.0.

File: tests/logging_helper_test.py (hunk 2 above)
Comment (mjbryant) [inline tests/logging_helper_test.py]: @kaisen these fake IDs will change to be "01d3120f591212" (16 char hex strings) in response to your PR.

**Comment by mjbryant** [comment]
BTW, I've tested this extensively in dev; I just didn't want to post links to those tests since this is a public github and they were very Yelp-specific.

**Pull request merged**